### ISSUE TITLE
Exception when attempting to migrate https://github.com/ReactiveX/rxdart

### ISSUE THREAD

**Issue by stereotype441** [dart-lang/sdk#43430]
@frankpepermans reports that trying run the migration tool on https://github.com/ReactiveX/rxdart produced the following exception:

```
RangeError (index): Invalid value: Valid value range is empty: 0

#0      List.[] (dart:core-patch/growable_array.dart:177:60)
#1      _AssignmentChecker._checkAssignment_recursion (package:nnbd_migration/src/edge_builder.dart:3304:51)
#2      _AssignmentChecker._checkAssignment (package:nnbd_migration/src/edge_builder.dart:3187:5)
#3      _AssignmentChecker._checkAssignment_recursion (package:nnbd_migration/src/edge_builder.dart:3302:9)
#4      _AssignmentChecker._checkAssignment (package:nnbd_migration/src/edge_builder.dart:3187:5)
#5      EdgeBuilder._handleAssignment (package:nnbd_migration/src/edge_builder.dart:2266:9)
#6      EdgeBuilder._handleInvocationArguments (package:nnbd_migration/src/edge_builder.dart:2790:7)
#7      EdgeBuilder.visitMethodInvocation (package:nnbd_migration/src/edge_builder.dart:1219:24)
#8      MethodInvocationImpl.accept (package:analyzer/src/dart/ast/ast.dart:7112:49)
#9      EdgeBuilder._dispatch (package:nnbd_migration/src/edge_builder.dart:2104:22)
#10     EdgeBuilder._handleAssignment (package:nnbd_migration/src/edge_builder.dart:2218:20)
#11     EdgeBuilder.visitReturnStatement (package:nnbd_migration/src/edge_builder.dart:1416:7)
#12     ReturnStatementImpl.accept (package:analyzer/src/dart/ast/ast.dart:8543:49)
#13     EdgeBuilder._dispatch (package:nnbd_migration/src/edge_builder.dart:2104:22)
#14     EdgeBuilder.visitNode (package:nnbd_migration/src/edge_builder.dart:1258:9)
#15     GeneralizingAstVisitor.visitStatement (package:analyzer/dart/ast/visitor.dart:517:39)
#16     GeneralizingAstVisitor.visitBlock (package:analyzer/dart/ast/visitor.dart:165:31)
#17     BlockImpl.accept (package:analyzer/src/dart/ast/ast.dart:1076:49)
#18     EdgeBuilder._dispatch (package:nnbd_migration/src/edge_builder.dart:2104:22)
#19     EdgeBuilder.visitNode (package:nnbd_migration/src/edge_builder.dart:1258:9)
#20     GeneralizingAstVisitor.visitFunctionBody (package:analyzer/dart/ast/visitor.dart:324:45)
#21     GeneralizingAstVisitor.visitBlockFunctionBody (package:analyzer/dart/ast/visitor.dart:168:55)
#22     BlockFunctionBodyImpl.accept (package:analyzer/src/dart/ast/ast.dart:1031:49)
#23     EdgeBuilder._dispatch (package:nnbd_migration/src/edge_builder.dart:2104:22)
#24     EdgeBuilder._handleExecutableDeclaration (package:nnbd_migration/src/edge_builder.dart:2346:7)
#25     EdgeBuilder.visitMethodDeclaration (package:nnbd_migration/src/edge_builder.dart:1177:5)
#26     MethodDeclarationImpl.accept (package:analyzer/src/dart/ast/ast.dart:6959:49)
#27     EdgeBuilder._dispatch (package:nnbd_migration/src/edge_builder.dart:2104:22)
#28     EdgeBuilder._dispatchList (package:nnbd_migration/src/edge_builder.dart:2118:7)
#29     EdgeBuilder.visitClassOrMixinOrExtensionDeclaration (package:nnbd_migration/src/edge_builder.dart:579:7)
#30     EdgeBuilder.visitExtensionDeclaration (package:nnbd_migration/src/edge_builder.dart:774:5)
#31     ExtensionDeclarationImpl.accept (package:analyzer/src/dart/ast/ast.dart:3814:49)
#32     EdgeBuilder._dispatch (package:nnbd_migration/src/edge_builder.dart:2104:22)
#33     EdgeBuilder.visitNode (package:nnbd_migration/src/edge_builder.dart:1258:9)
#34     GeneralizingAstVisitor.visitCompilationUnit (package:analyzer/dart/ast/visitor.dart:202:51)
#35     CompletenessTracker.visitCompilationUnit.<anonymous closure> (package:nnbd_migration/src/utilities/completeness_tracker.dart:52:24)
#36     PermissiveModeVisitor.reportExceptionsIfPermissive (package:nnbd_migration/src/utilities/permissive_mode.dart:26:24)
#37     CompletenessTracker.visitCompilationUnit (package:nnbd_migration/src/utilities/completeness_tracker.dart:43:5)
#38     CompilationUnitImpl.accept (package:analyzer/src/dart/ast/ast.dart:2119:49)
#39     NullabilityMigrationImpl.processInput (package:nnbd_migration/src/nullability_migration_impl.dart:190:12)
#40     NonNullableFix.processUnit (package:nnbd_migration/src/front_end/non_nullable_fix.dart:159:15)
#41     _FixCodeProcessor.runLaterPhases.<anonymous closure> (package:nnbd_migration/migration_cli.dart:1023:19)
#42     _FixCodeProcessor.processResources (package:nnbd_migration/migration_cli.dart:967:30)
<asynchronous suspension>
#43     _FixCodeProcessor.runLaterPhases (package:nnbd_migration/migration_cli.dart:1021:11)
#44     MigrationCliRunner.run (package:nnbd_migration/migration_cli.dart:610:43)
<asynchronous suspension>
#45     MigrateCommand.run (package:nnbd_migration/migration_cli.dart:204:72)
#46     CommandRunner.runCommand (package:args/command_runner.dart:197:27)
#47     DartdevRunner.runCommand (package:dartdev/dartdev.dart:324:24)
#48     CommandRunner.run.<anonymous closure> (package:args/command_runner.dart:112:25)
#49     new Future.sync (dart:async/future.dart:223:31)
#50     CommandRunner.run (package:args/command_runner.dart:112:14)
#51     runDartdev (package:dartdev/dartdev.dart:143:27)
#52     main (file:///C:/b/s/w/ir/cache/builder/sdk/pkg/dartdev/bin/dartdev.dart:11:9)
#53     _startIsolate.<anonymous closure> (dart:isolate-patch/isolate_patch.dart:297:32)
#54     _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:168:12)
```

I haven't had a chance to try to reproduce this yet, but I'm guessing that the starting point for migration was probably https://github.com/ReactiveX/rxdart/commit/ab85ab0e31c9ecd4644d5cd60985a056232d3633.

**Comment by stereotype441** [comment]
This is most likely the same issue as https://github.com/dart-lang/sdk/issues/44178, which was fixed in c0586277d78821e7c50c7e8aca507846e5844561, so I'm going to close as a duplicate.  If it happens again, please feel free to re-open.

**Comment by stereotype441** [comment]
Duplicate of #44178.
